File: Patterns/Abstract Factory Pattern/CSharp/Pizza/ClamPizza.cs
using System;

using PizzaIngredientFactory = Pizza.Ingredient.PizzaIngredientFactory;

namespace Pizza
{
    public class ClamPizza : Pizza
    {
        private PizzaIngredientFactory ingredientFactory;

        public ClamPizza(PizzaIngredientFactory ingredientFactory)
        {
            this.ingredientFactory = ingredientFactory;
        }

        public override void Prepare()
        {
            Console.WriteLine($"준비 중: {Name}");

            dough = ingredientFactory.CreateDough();
            sauce = ingredientFactory.CreateSauce();
            clams = ingredientFactory.CreateClams();

            Console.WriteLine($"{dough?.Name}을 뿌리는 중...");
            Console.WriteLine($"{sauce?.Name}을 뿌리는 중...");
            Console.WriteLine("토핑을 올리는 중...");
            Console.WriteLine($"\t{clams?.Name}");
        }
    }
}

File: Patterns/Abstract Factory Pattern/Cpp/Pizza/ClamPizza.cpp
#pragma once

#include "./Pizza.cpp"
#include "./Ingredient/PizzaIngredientFactory.cpp"

class ClamPizza : public Pizza {
public:
    PizzaIngredientFactory* ingredientFactory;

    ClamPizza(PizzaIngredientFactory* ingredientFactory) : ingredientFactory{ingredientFactory} {}

    void prepare() override {
        cout << "준비 중: " << name;

        dough = ingredientFactory->createDough();
        sauce = ingredientFactory->createSauce();
        clams = ingredientFactory->createClams();

        cout << dough->getName() << "를 돌리는 중...\n";
        cout << sauce->getName() << "를 뿌리는 중...\n";
        cout << "토핑을 올리는 중...\n";
        cout << '\t' << clams->getName() << '\n';
    }
};